OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyUtils.cs

```diff
@@ -464,8 +464,11 @@ public static void SetupReadyWidget(Widget parent, Session.Slot s, Session.Clien
 		public static void AddPlayerFlagAndName(ScrollItemWidget template, Player player)
 		{
 			var flag = template.Get<ImageWidget>("FLAG");
-			flag.GetImageName = () => player.Country.Race;
 			flag.GetImageCollection = () => "flags";
+			if (player.World.RenderPlayer != null && player.World.RenderPlayer.Stances[player] != Stance.Ally)
```

Comment:
IMO this way is better. As a spectator, you select to see the games through the eyes of player A. You should see what he sees and nothing more.